### PR TITLE
Bevy 0.9.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by fatfingers23**
Really appreciate this awesome plugin! Was wanting to make a Match 3 Bevy 0.11 game and thought i would submit a PR of what i did to get it running in 0.11 so others can use it too.

### CODE REVIEW

File: examples/basic.rs [hunk 1]
```diff
@@ -13,24 +16,24 @@ fn main() {
...
-        .add_system(move_to)
-        .add_system(consume_events)
-        .add_system(input)
-        .add_system(visualize_selection)
-        .add_system(control)
-        .add_system(animate_once)
-        .add_system(shuffle)
+        .add_plugins(Match3Plugin)
+        .add_systems(Startup, setup_graphics)
+        .add_systems(Update, move_to)
```
Comment (Sheepyhead) [inline examples/basic.rs]: This could be changed to a single ```.add_systems(Update, (move_to, ...))```

File: examples/basic.rs [hunk 2]
```diff
@@ -222,50 +227,32 @@ fn board_pos_to_world_pos(pos: &UVec2) -> Vec2 {
 struct Selection(Option<Entity>);
 
 fn input(
-    windows: Res<Windows>,
+    window_query: Query<&Window, With<PrimaryWindow>>,
     mut selection: ResMut<Selection>,
-    mut button_events: EventReader<MouseButtonInput>,
+    mut button_events: Res<Input<MouseButton>>,
```
Comment (Sheepyhead) [inline examples/basic.rs]: Why this change? Afaik button events should still work

File: examples/basic.rs [hunk 3]
```diff
@@ -222,50 +227,32 @@ fn board_pos_to_world_pos(pos: &UVec2) -> Vec2 {
...
-                    camera_transform.compute_matrix() * camera.projection_matrix().inverse();
-
-                // use it to convert ndc to world-space coordinates
-                let world_pos = ndc_to_world.project_point3(ndc.extend(-1.0));
-
-                // reduce it to a 2D value
-                let world_pos: Vec2 = world_pos.truncate();
-
-                // end of borrowed boilerplate
+    let window = window_query.get_single().unwrap();
```
Comment (Sheepyhead) [inline examples/basic.rs]: Instead of ```get_single().unwrap()``` you can simply use ```single()```

File: src/lib.rs [hunk 4]
```diff
@@ -59,7 +59,7 @@ impl Plugin for Match3Plugin {
         app.insert_resource(board)
             .insert_resource(BoardCommands::default())
             .insert_resource(BoardEvents::default())
-            .add_system(read_commands);
+            .add_systems(Update,read_commands);
```
Comment (Sheepyhead) [inline src/lib.rs]:
```suggestion
            .add_systems(Update, read_commands);
```

**Review by Sheepyhead — CHANGES_REQUESTED** [review]
Thank you so much for this pr! I'm very happy to see some interest in this crate, the only reason I haven't updated it is because nobody were really using it. I'll have to do a bit of manual testing to make sure it works as I expect, but I really appreciate the contribution! I only had a few nits

**Comment by Sheepyhead** [comment]
Also please make sure there are no warnings emitted

File: examples/basic.rs (hunk 2 above)
Comment (fatfingers23) [reply examples/basic.rs]: I'm still a bit new to rust and pretty sure I was fighting complier errors, and I found `Res<Input<MouseEvent>>` got what I needed. But I changed it back, and it should be good now.

**Comment by fatfingers23** [comment]
Should be the items you listed in that last PR. Please let me know if anything else needs to be changed or missed in that PR!

File: examples/basic.rs [hunk 5]
```diff
@@ -232,40 +236,27 @@ fn input(
         if let MouseButtonInput {
             button: MouseButton::Left,
             state: ButtonState::Pressed,
+            window: _
```
Comment (Sheepyhead) [inline examples/basic.rs]:
```suggestion
            ..
```

Comment (Sheepyhead) [reply examples/basic.rs]: This is more sustainable if more fields are added to this event later

File: src/board.rs [hunk 6]
```diff
@@ -323,7 +323,7 @@ impl BoardPosition for UVec2 {
 
 #[cfg(test)]
 mod tests {
-    use bevy::math::UVec2;
+    use bevy::math::{UVec2};
```
Comment (Sheepyhead) [inline src/board.rs]:
```suggestion
    use bevy::math::UVec2;
```

File: src/lib.rs [hunk 7]
```diff
@@ -122,21 +122,19 @@ mod tests {
             .add(BoardCommand::Swap([1, 2].into(), [2, 2].into()))
             .unwrap();
 
-        let mut update_stage = SystemStage::parallel();
-        update_stage.add_system(read_commands);
+        let mut update_stage = App::new();
```
Comment (Sheepyhead) [inline src/lib.rs]: Since `update_stage` is now no longer a stage but an app, it'd make sense to rename it to `app` in these tests

**Review by Sheepyhead — CHANGES_REQUESTED** [review]
I had a few more nits, but otherwise it looks great!

**Review by Sheepyhead — APPROVED** [review]
Thanks for the PR! I'll get a release out!
